README.md

```diff
@@ -1,2 +1,578 @@
+async functions:
+
+```javascript
+    async.auto({
+        'one': function (results, cb) {
+            // step one
+        },
+        'two': ['one', function (results, cb) {
+            // step two
+        }]
+    }, function (err, results) {
+        // all done
+    })
+```
+
+When dealing with complex controllers, you often find it easier to move the
+step implementations into their own functions:
+
+```javascript
+    function one(results, cb) {
+        // step one
+    }
+
+    function two(results, cb) {
+        // step two
+    }
+
+    async.auto({
+        'one': one,
+        'two': ['one', two]
+    }, function (err, results) {
+        // all done
+    })
+```
+
+This allows for easier unit testing and debugging and helps keep your code
+clean. But you may often find yourself adding lots of small wrappers to
+convert values between the various steps:
+
+```javascript
+    function one(results, cb) {
+        cb(null, {
+            'alpha': 'foo',
+            'beta': 'bar'
+        })
+    }
+
+    function two(results, cb) {
+        var params = {
+            'alpha': results.one.alpha,
+            'beta': results.one.beta
+        }
+
+        // step two
+    }
+```
+
+auto-block allows you to declare all of the mappings in the same place you
+declare the dependencies:
+
+```javascript
+    block = {
+        'one': {
+            func: one
+        },
+        'two': {
+            func: two,
+            with: {
+                'alpha': 'one.alpha',
+                'beta': 'one.beta'
+            }
+        }
+    }
+```
+
+Similarly, controllers often need to detect errors or attaching extra data if
+a step fails:
+
+```javascript
+    async.auto({
+        'one': utility.doStuff,
+        'two': function (results, cb) {
+            var options = {
+                'alpha': results.one.alpha,
+                'beta': results.one.beta
+            }
+            utility.findUser(options, function (err, result) {
+                if (err) {
+                    err.status = 404
+                    err.alpha = options.alpha
+                }
+                cb(err, result);
+            })
+        },
+        'two': ['one', function (results, cb) {
+            // step two
+        }]
+    }, function (err, results) {
+        if (err && err.status) {
+            res.status(err.status).send({
+                message: err.message
+            })
+        }
+    })
+```
+
+auto-block lets you declare these error mappings using `errorDefaults` and
+`errorMappings`:
+
+```javascript
+    var controller: {
+        block: {
+            'one': {
+                func: one
+            },
+            'two': {
+                func: two,
+                with: {
+                    'alpha': 'one.alpha',
+                    'beta': 'one.beta'
+                },
+                errorDefaults: {
+                    'status': 404
+                },
+                errorMappings: {
```

Comment:
Yeah, I pondered this for a while. The original reason for `errorMappings` and `errorDefaults` actually doesn't exist anymore so I'll try it this way and see how it feels.